frontend/scripts/componentSchemas/schemaUtils.ts

```diff
@@ -1,19 +1,119 @@
 import jsonpointer from 'jsonpointer';
+import { sortTextResourceBindings } from './languageUtils';
+import type {
+  LayoutSchema,
+  CondensedComponentSchema,
+  ExpandedComponentSchema,
+  AppFrontendVersion,
+  ComponentName,
+  PrefixedComponentName,
+} from './types';
+import type { KeyValuePairs } from 'app-shared/types/KeyValuePairs';
+
+export const allPropertyKeys = [];
+
+export const generateComponentSchema = (
+  componentName: ComponentName,
+  layoutSchema: LayoutSchema,
+  version: AppFrontendVersion,
+): ExpandedComponentSchema => {
+  const definitionName: PrefixedComponentName = `Comp${componentName}`;
+  const condensedComponentSchema = getCondensedComponentSchema(
+    definitionName,
+    layoutSchema,
+    version,
+  );
+  let newComponentSchema = initializeComponentSchema(componentName, layoutSchema);
+
+  newComponentSchema = expandSchema(condensedComponentSchema, newComponentSchema, layoutSchema);
+  sortSchemaTextResourceBindings(newComponentSchema);
+  newComponentSchema.title = `${componentName} component schema`;
+
+  return newComponentSchema as ExpandedComponentSchema;
+};
+
+export const getCondensedComponentSchema = (
+  definitionName: PrefixedComponentName,
+  layoutSchema: LayoutSchema,
+  version: AppFrontendVersion,
+): CondensedComponentSchema => {
+  if (version === 'v4') {
+    console.log('definitionName: ', definitionName + 'External');
+    return expandRef(
+      layoutSchema.definitions[definitionName].$ref,
+      layoutSchema,
+    ) as CondensedComponentSchema;
+  }
+  console.log('definitionName: ', definitionName);
+  return layoutSchema.definitions[definitionName] as CondensedComponentSchema;
+};
+
+const initializeComponentSchema = (
+  componentName: ComponentName,
+  layoutSchema: LayoutSchema,
+): Partial<ExpandedComponentSchema> => {
+  return {
+    $id: `https://altinncdn.no/schemas/json/component/${componentName}.schema.v1.json`,
+    $schema: layoutSchema.$schema,
+  };
+};
+
+const expandSchema = (
+  condensedSchema: CondensedComponentSchema,
+  expandedSchema: Partial<ExpandedComponentSchema>,
+  layoutSchema: LayoutSchema,
+): Partial<ExpandedComponentSchema> => {
+  if (condensedSchema.allOf) {
+    expandedSchema = { ...expandedSchema, ...expandAllOf(condensedSchema, layoutSchema) };
+
+    const expectedProperties = getExpectedProperties(condensedSchema);
+    allPropertyKeys.push(...expectedProperties);
+
+    if (!verifySchema(expandedSchema, expectedProperties)) {
+      return null;
+    }
+
+    // This else if does not work in the current configuration. Therefore, the line inside is commented out.
+    // It seems to only apply to the Group component in frontend v3, which we may not maintain much longer.
+  } else if (condensedSchema.anyOf) {
+    // newSchema.anyOf = expandAnyOf(condensedSchema, layoutSchema);
+  }
```

Comment:
After adding typechecking, I found out that this part of the script is broken.
There should not be an `anyOf` property at the root level of the emitted component schema. This only seems to apply to the Group component in frontend v3, which may not be maintained much longer (?)

I tried to fix it, but could not make heads or tails of it right now. I suggest we create a new issue for this.

Reply:
@ErlingHauan, I think it is a good idea that you create an issue for it 😄 
Can you also add a comment in the code referring to the issue with an url, so that the next person in the code can see that there is an issue for it?
v3 is not much used any more, but we still support it and should be able to do fixes on it as long as apps says it is being support 😄

Reply:
I did add a link to the [new issue](https://github.com/Altinn/altinn-studio/issues/13785) in the code 😉